File: 509.fibonacci-number.py
#
# @lc app=leetcode id=509 lang=python3
#
# [509] Fibonacci Number
#

# @lc code=start
class Solution:
    def fib(self, N: int) -> int:
        if N == 0:
            return 0
        res = [0,1]
        for i in range(2,N+1):
            res.append(res[-1]+res[-2])
        return res[-1]
        
# @lc code=end

File: golang/509.fibonacci-number.go
/*
 * @lc app=leetcode.cn id=509 lang=golang
 *
 * [509] Fibonacci Number
 */
package main

// @lc code=start
func fib(n int) int {
	if n == 0 {
		return 0
	}
	dp := []int{0, 1}
	for i := 2; i <= n; i++ {
		dp = append(dp, dp[i-1]+dp[i-2])
	}
	return dp[len(dp)-1]
}

// @lc code=end
